### DISCUSSION TITLE
Override .meta.yml on individual pages in the folder?

### DISCUSSION THREAD

**Discussion by rachel-rigdon-sp** [squidfunk/mkdocs-material#6913] [Q&A] [answered]
I am looking for help to override the .meta.yml configuration for some pages within the folder. 

Context: 

I am adding a Discourse comments section to our documentation. I've set the .meta.yml to `comments: true` and placed it in our docs/ folder, which successfully applied the comments.html override with the Discourse code to all pages in the folder.

However, there are some pages within the folder (like our main index.md) that we don't want comments on. I've tried excluding them using meta formatting at the top, like: 
```
---
comments: false
---
```
I've also tried `comments: hide`, `comments: exclude` on the individual page, to no avail. 

Is there a way to override the .meta.yml on some pages within the shared folder?

**Comment by kamilkrzyskow** [comment]
Hello @rachel-rigdon-sp,
I feel like your reasoning is just and the front matter meta of a page should override the defaults in the meta.yml file.

However, the current implementation merges the defaults onto the page's meta:
https://github.com/squidfunk/mkdocs-material-insiders/blob/master/material%2Fplugins%2Fmeta%2Fplugin.py#L93-L95

I'm unfamiliar with merge strategies, so perhaps the intention of additive was to only merge values that are not present, but `false` is replaced too.

A quick solution would be to make a hook to replace the value after the meta plugin already processed the page.

https://github.com/squidfunk/mkdocs-material/discussions/6324#discussioncomment-7550511
~~Modify the page.meta value assignment to `comments` and `False` you can also delete the blog part.~~ Read comment below 👇

Apart of that please create a bug report.

**Reply by rachel-rigdon-sp** [reply]
Thank you!

**Reply by kamilkrzyskow** [reply]
I'm away from my PC today, so the comment about the hook was too simplified out of laziness.

~~Here is a better example:~~ Incorrect, read comment below 👇 

<details>

```py
def on_page_markdown(markdown, *, page, config, files): 
    comments = page.meta.get("comments")
    if comments is not None:
        page.meta["comments"] = comments
```

This will allow you to override it in the front matter as you wanted

</details>

**Reply by kamilkrzyskow** [reply]
So after pondering over this a bit further, the example won't work, as using `page.meta` after the meta plugin finishes processing will lead to using the already replaced value 🤦 😵 💫 Sorry for this rookie mistake 😞 

Here is an actually tested example, 3rd attempt 🙌:
[meta-fix.zip](https://github.com/squidfunk/mkdocs-material/files/14611548/meta-fix.zip) (example uses the `status` front matter, as setting up a comment system for the example seemed unnecessary)
```py
from copy import deepcopy
from typing import Any, MutableMapping

from mkdocs import plugins
from mkdocs.structure.pages import Page

FILE_AND_META: dict[str, MutableMapping[str, Any]] = {}


@plugins.event_priority(100)
def on_config(*_):
    global FILE_AND_META
    FILE_AND_META = {}


@plugins.event_priority(100)
def on_page_markdown(*_, page: Page, **__):
    FILE_AND_META[page.file.src_uri] = deepcopy(page.meta)


@plugins.event_priority(100)
def on_page_content(*_, page: Page, **__):
    page.meta = {**page.meta, **FILE_AND_META[page.file.src_uri]}

```

The meta plugin uses the `50` priority value, so the hook has to run faster and copy the initial meta, then the actual meta fix is in the next event after the meta plugin done the work. 

EDIT: Made small edit to support `mkdocs serve`

**Comment by squidfunk** [comment, marked as answer]
This turned out to be a bug (read: unintended behavior) which was reported in #6914 and fixed.
